CEXT-4103: Document oope-shipping-methods module (#315)
AdobeDocs/commerce-extensibility: src/pages/starter-kit/checkout/use-cases.md

The following example demonstrates how to add a webhook to the `plugin.magento.out_of_process_shipping_methods.api.shipping_rate_repository.get_rates` method:

```xml
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_AdobeCommerceWebhooks:etc/webhooks.xsd">
    <method name="plugin.magento.out_of_process_shipping_methods.api.shipping_rate_repository.get_rates" type="after">
        <hooks>
            <batch name="dps">
                <hook name="add_shipping_rates_dps" url="https://<yourappbuilder>.runtime.adobe.io/api/v1/web/commerce-checkout-starter-kit/shipping-methods" method="POST" timeout="5000" softTimeout="1000" priority="100" required="true">
                    <fields>
                        <field name="rateRequest" />
                    </fields>
                </hook>
            </batch>
        </hooks>
    </method>
</config>
```

Or to navigate to **System > Webhooks** in the Adobe Commerce Admin and create a new webhook with the following configuration:

> ```

Author: Added

Hook Settings
  Webhook Method: plugin.magento.out_of_process_shipping_methods.api.shipping_rate_repository.get_rates
  Webhook Type: after
  Batch Name shipping_methods
  Hook Name: oope_shipping_methods_carrier_one
  URL: https://<yourappbuilder>.runtime.adobe.io/api/v1/web/commerce-checkout-starter-kit/shipping-methods
  Active: Yes
  Method: POST

Hook Fields
  Field: rateRequest
```

You can register multiple webhooks for different shipping methods or shipping carriers by adding them into the same batch to ensure they are executed in parallel or create multiple batches to execute them sequentially.

## Shipping methods: GraphQL

> In the `setShippingAddressesOnCart` the available shipping methods that are returned by the webhook are appended to the `available_shipping_methods` field.

Reviewer: ```suggestion
In `setShippingAddressesOnCart` available shipping methods that are returned by the webhook are appended to the `available_shipping_methods` field.
```

> You can use the `additional_data` field to pass additional information about the shipping method from the webhook. The `additional_data` field is an array of key-value pairs that can be useful for passing additional information about the shipping method.

Reviewer: ```suggestion
You can use the `additional_data` field to pass an array of key-value pairs to provide additional information about the shipping method from the webhook.```

                    "value": "additional_data_value"
                  },
                  {
                    "key": "additional_data_key2",
                    "value": "additional_data_value2"
                  },
                  {
                    "key": "additional_data_key3",
                    "value": "additional_data_value3"
                  }
                ]
              },
              {
                "amount": {
                  "currency": "USD",
                  "value": 18
                },
                "available": true,
                "carrier_code": "DPS",
                "carrier_title": "Demo Postal Service",
                "error_message": "",
                "method_code": "dps_shipping_two",
                "method_title": "Demo Custom Shipping Two",
                "price_excl_tax": {
                  "currency": "USD",
                  "value": 18
                },
                "price_incl_tax": {
                  "currency": "USD",
                  "value": 18
                },
                "additional_data": []
              }
            ]
          }
        ]
      }
    }
  }
}
```

> In the `setShippingMethodsOnCart` mutation you can set the shipping method provided by webhook, it's information will be stored in the `selected_shipping_method` field with the `additional_data` if provided.

Reviewer: ```suggestion
In the `setShippingMethodsOnCart` mutation you can set the shipping method provided by webhook, its information is stored in the `selected_shipping_method` field with the `additional_data` if provided.
```
